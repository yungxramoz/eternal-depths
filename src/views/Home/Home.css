@keyframes lightning {
  from {
    filter: brightness(100%);
  }
  92% {
    filter: brightness(100%);
  }
  93% {
    filter: brightness(40%);
  }
  94% {
    filter: brightness(80%);
  }
  96% {
    filter: brightness(10%);
  }
  to {
    filter: brightness(100%);
  }
}

h1.main-title {
  text-align: center;
  font-size: 2rem;
  line-height: 2.5rem;
}

.button-menu {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-top: 2rem;
}

.button-menu button {
  width: 100%;
}

.home-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  /* animation: lightning ease-out 10s infinite; */
  animation-delay: 2s;
}
