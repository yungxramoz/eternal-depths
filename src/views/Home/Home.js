import { Link } from 'react-router-dom'
import RpgButton from '../../components/atoms/RpgButton/RpgButton'
import RpgContainer from '../../components/templates/RpgContainer/RpgContainer'
import RpgSeparator from '../../components/templates/RpgSeperator/RpgSeparator'
import './Home.css'

import bgImg from '../../assets/images/ui/home.png'

function Home() {
  return (
    <RpgContainer fullPage className="home-container" bgImg={bgImg}>
      <h1 className="main-title">Eternal Depths</h1>
      <RpgSeparator />
      <div className="button-menu">
        <Link to="/new-character">
          <RpgButton text="New Character" />
        </Link>
        <Link to="/leaderboard">
          <RpgButton text="Leaderboard" />
        </Link>
      </div>
    </RpgContainer>
  )
}

export default Home
