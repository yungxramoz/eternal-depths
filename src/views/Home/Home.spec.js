import React from 'react'
import TestRenderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import Home from './Home'

describe('Home', () => {
  test('renders correctly', () => {
    const tree = TestRenderer.create(
      <MemoryRouter>
        <Home />
      </MemoryRouter>,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('contains correct elements and text', async () => {
    const testRenderer = TestRenderer.create(
      <MemoryRouter>
        <Home />
      </MemoryRouter>,
    )
    const testInstance = testRenderer.root
    const h1 = await testInstance.findByType('h1')
    expect(h1.children).toEqual(['Eternal Depths'])
    const buttons = await testInstance.findByProps({ className: 'button-menu' })
    expect(buttons.children[0].props.to).toEqual('/new-character')
    expect(buttons.children[0].props.children.props.text).toEqual(
      'New Character',
    )
    expect(buttons.children[1].props.to).toEqual('/leaderboard')
    expect(buttons.children[1].props.children.props.text).toEqual('Leaderboard')
  })
})
