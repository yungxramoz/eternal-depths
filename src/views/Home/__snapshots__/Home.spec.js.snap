// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="rpgui-container framed full-page home-container"
  style={
    Object {
      "backgroundImage": "linear-gradient( rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5) ), url(home.png)",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
    }
  }
>
  <h1
    className="main-title"
  >
    Eternal Depths
  </h1>
  <hr
    className="separator golden"
  />
  <div
    className="button-menu"
  >
    <a
      href="/new-character"
      onClick={[Function]}
    >
      <button
        className="rpgui-button btn"
      >
        New Character
      </button>
    </a>
    <a
      href="/leaderboard"
      onClick={[Function]}
    >
      <button
        className="rpgui-button btn"
      >
        Leaderboard
      </button>
    </a>
  </div>
</div>
`;
